validation/src/blockchain/eth_rpc.rs

```diff
@@ -0,0 +1,295 @@
+use std::{
+    num::NonZero,
+    str::FromStr,
+    sync::{Arc, Mutex},
+};
+
+use anyhow::{anyhow, Context, Result};
+use lru::LruCache;
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+use ssi::caip2;
+use tracing::trace;
+
+const TRANSACTION_CACHE_SIZE: usize = 50;
+const BLOCK_CACHE_SIZE: usize = 50;
+
+static HTTP_CLIENT: Lazy<reqwest::Client> = Lazy::new(reqwest::Client::new);
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ChainTransaction {
+    /// Transaction hash
+    pub hash: String,
+    /// Transaction contract input
+    pub input: String,
+    pub block: Option<BlockResponse>,
```

Comment:
it can be done if the transaction hasn't been mined yet.. not sure if we actually want to deal with that or just error, but the current code has these cases:

1. transaction exists but not mined (no timestamp yet)
2. block mined -> timestamp available
3. transaction not found (may be an error, may just not exist on our provider for some reason)
4. error communicating with rpc endpoint

Reply:
Makes sense.  I think a simple comment saying it's None if the txn hasn't been mined yet would be fine